src/components/context.js

```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+
+const AppContext = React.createContext();
+
+export class ContextProvider extends React.Component {
+  render() {
+    return (
+      <AppContext.Provider value={this.props.markdownRenderer}>
+        {this.props.children}
+      </AppContext.Provider>
+    );
+  }
+}
+
+export class ContextConsumer extends React.Component {
+  render() {
+    return (
+      <AppContext.Consumer>
+        {markdownRenderer => this.props.children(markdownRenderer)}
+      </AppContext.Consumer>
+    );
+  }
+}
```

Comment:
A shorter, cleaner, type-safe implementation is possible here. Checkout this example:

https://github.com/apimatic/apimatic-dx-portal/blob/9685e9af2e5de5499d4b7ae237a7e06c81908d14/src/PortalContext.tsx#L8-L14

Reply:
Done